ecleankernel/layout/blspec.py

```diff
@@ -119,6 +124,23 @@ def find_kernels(self,
                         k, ver, module_dict, exclusions)
                 kernels[ver].all_files.append(EmptyDirectory(dir_path))
 
+        # collect from ESP/Linux
+        if self.ukidir.is_dir():
+            for file in os.listdir(self.ukidir):
+                if (not file.startswith(self.kernel_id) or
+                        not file.endswith(".efi")):
+                    # This file is not an efi file or does not belong to us
+                    continue
+                ver = file.removeprefix(self.kernel_id +
+                                        "-").removesuffix(".efi") + " <UKI> "
```

Comment:
Honestly, I keep wondering whether there's a "better way", like one that doesn't check for the prefix twice. However, I can't think of anything good:

- `file[len(self.kernel_id)+1:-4]` is kinda ugly
- splitting into three vars and checking if `remove*` removed anything every time kinda makes you think of poor man's shell scripting
- regex would be just… meh.

But why are you adding a space *after* `<UKI>`?

Reply:
> But why are you adding a space after <UKI>?

Just a precaution to ensure that when we `removesuffix` we don't have any false positives. Though probably the `<>` is already enough to ensure this doesn't happen.